aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -247,6 +247,13 @@ public static ModifyDbInstanceRequest modifyDbInstanceRequest(
             builder.iops(desiredModel.getIops());
         }
 
+        // In order to disable an instance autoscaling one has to reset `MaxAllocatedStorage` property.
+        // `MaxAllocatedStorage` would be reset upon an update if it is set to the current `AllocatedStorage` value.
+        // https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PIOPS.StorageTypes.html#USER_PIOPS.Autoscaling
+        if (previousModel != null && previousModel.getMaxAllocatedStorage() != null && desiredModel.getMaxAllocatedStorage() == null) {
+            builder.maxAllocatedStorage(getAllocatedStorage(previousModel));
```

Comment:
Could you elaborate why setting `maxAllocatedStorage` to previousModel `AllocatedStorage` not the desiredModel 
i.e:              `builder.maxAllocatedStorage(getAllocatedStorage(desiredModel)); `

Reply:
In fact, we need to pick up the current value, hence neither previous nor desired resource state is a trustful source of this information, as autoscaling might has already changed the current `AllocatedStorage` value. It needs to be picked up from an explicit describe call. I'm going to address it in a jiffy.